File: exam_app/lib/models/question.dart
import 'package:json_annotation/json_annotation.dart';
import 'answer.dart';

part 'question.g.dart';

@JsonSerializable()
class Question {
  final int id;
  final int? examId;
  final String questionText;
  final QuestionType type;
  final int orderIndex;
  final int points;
  final List<Answer>? answers;

  Question({
    required this.id,
    this.examId,
    required this.questionText,
    required this.type,
    required this.orderIndex,
    required this.points,
    this.answers,
  });

  factory Question.fromJson(Map<String, dynamic> json) => _$QuestionFromJson(json);
  Map<String, dynamic> toJson() => _$QuestionToJson(this);
}

enum QuestionType {
  @JsonValue('MULTIPLE_CHOICE')
  multipleChoice,
  @JsonValue('TRUE_FALSE')
  trueFalse,
  @JsonValue('SHORT_ANSWER')
  shortAnswer,
}

File: exam-api/src/main/java/com/camoleze/examapi/model/Question.java
package com.camoleze.examapi.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("questions")
public class Question {
    @Id
    private Long id;
    private Long examId;
    private String questionText;
    private QuestionType type;
    private Integer orderIndex;
    private Integer points;
    
    public enum QuestionType {
        MULTIPLE_CHOICE, TRUE_FALSE, SHORT_ANSWER
    }
}